blockchain/message_pool/src/msgpool.rs

```diff
@@ -560,35 +616,79 @@ where
 
         Ok(())
     }
+    /// If `local = true`, the local messages will be removed as well as pending messages.
+    /// If `local = false`, pending messages will be removed while retaining local messages.
+    pub async fn clear(&mut self, local: bool) {
+        if local {
+            let local_addrs = self.local_addrs.read().await;
+            for a in local_addrs.iter() {
+                if let Some(mset) = self.pending.read().await.get(&a) {
+                    for m in mset.msgs.values() {
+                        if !self.local_msgs.write().await.remove(&m) {
+                            warn!("error deleting local message");
+                        }
+                    }
+                }
+            }
+            self.pending.write().await.clear();
+        } else {
+            let mut pending = self.pending.write().await;
+            let local_addrs = self.local_addrs.read().await;
+            pending.retain(|a, _| local_addrs.contains(&a));
+        }
+    }
+    pub fn get_config(&self) -> &MpoolConfig {
+        &self.config
+    }
+    pub fn set_config<DB: Store>(&mut self, db: &DB, cfg: MpoolConfig) -> Result<(), Error> {
+        cfg.save_config(db)
+            .map_err(|e| Error::Other(e.to_string()))?;
+        self.config = cfg;
+        Ok(())
+    }
+}
+
+fn verify_msg_before_add(m: &SignedMessage, cur_ts: &Tipset, local: bool) -> Result<bool, Error> {
+    let epoch = cur_ts.epoch();
+    let min_gas = interpreter::price_list_by_epoch(epoch).on_chain_message(m.marshal_cbor()?.len());
+    m.valid_for_block_inclusion(min_gas.total())
+        .map_err(Error::Other)?;
+    if !cur_ts.blocks().is_empty() {
+        let base_fee = cur_ts.blocks()[0].parent_base_fee();
+        let base_fee_lower_bound = base_fee / BASE_FEE_LOWER_BOUND_FACTOR;
```

Comment:
They use the func getBaseFeeLowerBound when getting the lower bound in Lotus, if you want to keep it a one  liner tho, something this would be like,  let base_fee_lower_bound = max(base_fee / BASE_FEE_LOWER_BOUND_FACTOR, MINIMUM_BASE_FEE)